alternatives.md

```diff
@@ -0,0 +1,113 @@
+
+<!-- TODO: https://github.com/oracle/opengrok/wiki/Comparison-with-Similar-Tools -->
+
+## [Kythe](https://kythe.io/)
+
+> The best way to view Kythe is as a “hub” for connecting tools for various languages, clients and build systems. By defining language-agnostic protocols and data formats for representing, accessing and querying source code information as data, Kythe allows language analysis and indexing to be run as services.
+
+> Some tools (e.g., static analyzers) already have expressive purpose-built internal representations for code. Kythe is not meant to be a universal replacement for such IRs — instead, our goal is to provide a way for such tools to capture “interesting subsets” of an analysis for sharing with other tools.
+
+Babelfish and Kythe share a goal of defining a common representation
+for concepts from different programming languages and provide a way to query it.
+Both provide a unified data format. And both are language-independent.
+
+The main difference is that Babelfish preserves all AST nodes including
+control flow and expressions, while Kythe focuses on class hierarchy,
+dependencies, etc.
+
+Also, Babelfish provides a unified IR across languages, which is defined
+as a non-goal of Kythe.
+
+Kythe requires to instrument language compilers and build systems while
+Babelfish uses native language parsers to get an AST, allowing to develop
+a new language driver in less time.
+
+Kythe also provides a way to process the whole project, while for now
+Babelfish is focused on processing individual files.
+
+## [Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
+
+> A Language Server is meant to provide the language-specific smarts and communicate with development tools over a protocol that enables inter-process communication.
+
+> The idea behind the Language Server Protocol (LSP) is to standardize the protocol for how such servers and development tools communicate. This way, a single Language Server can be re-used in multiple development tools, which in turn can support multiple languages with minimal effort.
+
+LSP defines a common protocol and RPC for queries like Go-To-Definition,
+Usages, etc. But it does not define a common representation of AST because
+the goal of the project is to enable easy access to analysis that is done
+by compilers and language SDKs.
+
+Babelfish provides a common representation for ASTs allowing
+to use its output for static analysis by other tools. Queries over UAST
+structure will still allow querying for usages, etc.
+
+## [srclib](https://srclib.org/)
+
+> srclib makes developer tools like code search and static analyzers better. It supports things like jump to definition, find usages, type inference, and documentation generation.
+
+> srclib handles: package detection, global dependency resolution, type inference, querying the graph of definitions and references in code, versioning using different VCS systems, and semantic blaming.
+
+<!-- TODO -->
```

Comment:
AFAIK this project is not maintained by sourcegraph any more.

But I think it might worth spend a little bit of time here to highlight the difference, as one can see other companies in source code analysis field are using it i.e https://fossa.io/ though https://github.com/fossas/srclib and it would be nice to understand why bblfsh would not feat for them.

Reply:
AFAIK the main reseason FOSSA uses srclib and would not be able to use Babelfish is the dependency graph it can generate through symbolic cross-references (sorry for being a one trick pony answering this everywhere but it seems relevant).

Reply:
Isn't maintained. The abandoned in favor of LSP, AFAIK